pkg/util/errors/errors.go

```diff
@@ -29,10 +29,16 @@ type Aggregate interface {
 // is itself an implementation of the error interface.  If the slice is empty,
 // this returns nil.
 func NewAggregate(errlist []error) Aggregate {
-	if len(errlist) == 0 {
+	errs := []error{}
+	for _, err := range errlist {
+		if err != nil {
```

Comment:
I think my point was that a nil error should never find its way into an aggregate, and if it does it was caller's error.  We should not filter them out, though we might make them more obvious (if they occur, but afaik it has not happened)